TestResultSummaryService/plugins/DataManagerPerf.js

```diff
@@ -0,0 +1,65 @@
+const ObjectID = require( 'mongodb' ).ObjectID;
+const BenchmarkMath = require( '../perf/BenchmarkMathCalculation' );
+const math = require('mathjs');
+
+module.exports.onBuildDone = async (task, { testResultsDB, logger }) => {
+    logger.debug("onBuildDone", task.buildName);
+    if ( task.hasChildren ) {
+        const parentId = task._id;
+        const childBuildList = await testResultsDB.getData({parentId}).toArray();
+        if ( childBuildList && childBuildList.length > 0 ) {
+            const parentAggregationStructure = [];
+            let benchmarkName, benchmarkVariant, benchmarkProduct;
+            const childrenAggData = [];
+            const metricsCollection = {};
+            //loop into the child build list
+            for ( let j = 0; j < childBuildList.length; j++ ) {
+                if (childBuildList[j].aggregateInfo && childBuildList[j].aggregateInfo.length > 0) {
+                    //loop into the each child's aggregate info.
+                    for (let k = 0 ; k < childBuildList[j].aggregateInfo.length; k++){
```

Comment:
Same for the aggregateInfo loop below and the similar cases :)
Line 17 - 23 can be written as following:
```
if (Array.isArray(childBuild.aggregateInfo)) {
   for ( let {benchmarkName, benchmarkVariant, benchmarkProduct, metrics} of childBuild.aggregateInfo){
      if (Array.isArray(metrics)) {
```

Reply:
updated